core/views.py

```diff
@@ -583,3 +589,37 @@ def get_context_data(self, **kwargs):
         ctx['results'] = results
 
         return ctx
+
+class OpenJobListView(APIView):
+    def get(self, request):
+        # Parameters recieving
+        limit = int(request.query_params.get('limit', 10))
+        offset = int(request.query_params.get('offset', 0))
+
+        url = "https://jobs.ucf.edu/jobs/search"
+        response = requests.get(url)
```

Comment:
This is an area we probably want to code a little more protectively around. The default timeout for `requests.get` is `None`, which will cause the process to hang indefinitely if the web server doesn't respond with an error. This is a pretty common issue.

Sometimes you will get an error, and `requests.get` may throw an exception. We should probably try to handle this in our code by catching the exception and returning a specific response with an error HTTP code (most likely 500). Here's an example of that below.

```python
try:
   response = requests.get(url)
except:
   return Response("An error occurred fetching the jobs", status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Sometimes, the web server returns an error. We can double check on these by checking that the response is ok, using the convenient `if response.ok:` property that requests provides. If the response isn't okay, we should probably short circuit the rest of the code so BeautifulSoup doesn't try to parse an error page (or possibly an empty response).

Reply:
added the error handling to the code.